src/main/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapper.java

```diff
@@ -254,25 +250,27 @@ public GeoPointFieldMapper build(BuilderContext context) {
                     }
                     iterator.remove();
                 } else if (fieldName.equals("validate")) {
-                    builder.validateLat = XContentMapValues.nodeBooleanValue(fieldNode);
-                    builder.validateLon = XContentMapValues.nodeBooleanValue(fieldNode);
+                    builder.validate = XContentMapValues.nodeBooleanValue(fieldNode);
                     iterator.remove();
                 } else if (fieldName.equals("validate_lon")) {
-                    builder.validateLon = XContentMapValues.nodeBooleanValue(fieldNode);
-                    iterator.remove();
+                    if (parserContext.indexVersionCreated().onOrAfter(Version.V_1_6_0)) {
```

Comment:
Given how we've done other deprecations and removals, I think we need to maintain backcompat for at least reading and ignoring these settings in 2.0? Also, I think it is better to add move the version check to the `else if` and invert, then have a comment like `// ignore legacy setting` inside the block.  For newer indexes, the setting would fall through and show up as an unknown setting.